### ISSUE TITLE
Using {@@} for LaTeX causes the error "append: operators of arguments must all be the same."

### ISSUE THREAD

**Issue by helmo-nico** [maths/moodle-qtype_stack#1014]
Hello,

We have recently updated the Stack plugin to version 2023060500 on our Moodle (Moodle 4.1.4+ (Build: 20230804)). An instructor has reported a bug to me regarding LaTeX notation on one of their questions.

The issue affects STACK questions where two conditions are met:

- The PRT (Potential Response Tree) constructs a feedback message in multiple parts, meaning that several nodes traversed in the PRT generate messages. All of these messages are concatenated to obtain the final feedback.

- One of the messages uses the notation {@@}, which evaluates the expression using Maxima and generates LaTeX output (necessary if the evaluation result is a mathematical formula).

In this situation, an error message is produced as shown below, and the evaluation of the PRT is interrupted:
_**`The node {$a->node} generated the following error: append: operators of arguments must all be the same. append: operators of arguments must all be the same. append: operators of arguments must all be the same. append: operators of arguments must all be the same. append: operators of arguments must all be the same. append: operators of arguments must all be the same. append: operators of arguments must all be the same. append: operators of arguments must all be the same. append: operators of arguments must all be the same.`**_

The "append" error is repeated for each node contributing to the feedback.

The workaround he founds is to use the {##} notation instead of the other one. This results in plain text rather than LaTeX. In the case of the questions affected by this issue, it's not problematic since the returned value is a simple number, not a mathematical formula.

However, we believe this is a bug that should be reported, as this problem was not present in the previous deployed version.

If you wish to reproduce the problem, I am attaching the XML export of the question. To highlight the issue:

1. Choose the question with the bug and select "Preview."
2. Click on "Fill correct answers," then modify values in several columns.
3. Click on "Submit and finish" to evaluate the question.
![Capture-erreur-stack](https://github.com/maths/moodle-qtype_stack/assets/73701015/8184a21f-c943-4a32-8488-5bca44c8a11b)
[quiz-Testing SCHUF-Cache à correspondance directe (version avec le bug)-20230816-0921.zip](https://github.com/maths/moodle-qtype_stack/files/12357127/quiz-Testing.SCHUF-Cache.a.correspondance.directe.version.avec.le.bug.-20230816-0921.zip)

**Comment by aharjula** [comment]
Given that, for me things look like this:

![Screenshot from 2023-08-16 11-26-37](https://github.com/maths/moodle-qtype_stack/assets/980957/66332aa7-62e3-41fe-b4e5-3e423c6612fe)

I do have to make some assumptions, and the first that comes to mind is that when updating things, the update was not complete. There might be old STACK-Maxima libraries running somewhere which produce slightly different logic, which mismatch with the new Moodle side logic.

Do check the [healthcheck](https://docs.stack-assessment.org/en/Installation/Testing_installation/) script and especially if it states that the Maxima version matches, and if you happen to run some server Maxima solution like MaximaPool, you might need to update those scripts on that end, if not running a server setup then recreating any images might be a step that has been missed. If this is not the case then things are more complicated as I have trouble imagining other reasons, might be that I come up with some though.

**Comment by aharjula** [comment]
Given that that particular error message is about an `append` that happens on the Maxima side, it could be that the version is relevant. However, I do not remember whether `append` itself could have changed between those versions.

Would recommend updating to something like 5.44.0, that version has served me well for a while.

If you have CLI access to the Moodle server and are willing to test something before that, you could ensure that an extra cache gets cleared by calling:

```
php question/type/stack/cli/clearcascache.php --ccc
```
The extra `--ccc` clears something that should only be cleared if there has been an issue with the execution of the plugin upgrade scripts or if one does dev work.

**Comment by helmo-nico** [comment]
Unfortunately, this operation did not resolve the situation. We will schedule an update of Maxima to version 5.44.0 and the Stack plugin to its latest version. However, we need to wait until the end of the exam period, so this will be in a little less than a month. Thank you for your assistance; I will make sure to come back here and keep you informed of the progress.

**Comment by aharjula** [comment]
For completeness sake, I did test the attached question on a 2023060500 + 5.44.0 installation, and it did work. So the assumption for now is that the Maxima version is important for this part of logic, after all possible caches have been cleared and that as a possible reason has been eliminated. 

Have not tested with 5.38.1 as installing it on my new dev platform was not directly possible (the configuration scripts did not identify the system correctly).

Also, when we lifted the version requirements for Maxima, many kludges for dealing with older versions were cleaned from the codebase so it may have broken elsewhere as well if you are running non-supported versions.
